src/ufixed.jl

```diff
@@ -80,20 +80,12 @@ sizeof{T<:UFixed}(::Type{T}) = sizeof(rawtype(T))
 abs(x::UFixed) = x
 
 # Arithmetic
-(-){T<:UFixed}(x::T) = T(-reinterpret(x), 0)
-(~){T<:UFixed}(x::T) = T(~reinterpret(x), 0)
-
-+{T,f}(x::UFixed{T,f}, y::UFixed{T,f}) = UFixed{T,f}(convert(T, x.i+y.i),0)
--{T,f}(x::UFixed{T,f}, y::UFixed{T,f}) = UFixed{T,f}(convert(T, x.i-y.i),0)
-*{T,f}(x::UFixed{T,f}, y::UFixed{T,f}) = UFixed{T,f}((Base.widemul(x.i,y.i) + (convert(widen(T), 1) << (f-1) ))>>f,0)
-/{T,f}(x::UFixed{T,f}, y::UFixed{T,f}) = UFixed{T,f}(div(convert(widen(T), x.i)<<f, y.i),0)
+*{T<:UFixed}(x::T, y::T) = T(widemul(x[], y[]) ÷ rawone(T), 0)
```

Comment:
Ironic, though, that after touting the importance of using multiplication rather than division for operations, I just tracked down https://github.com/timholy/Images.jl/commit/e8d97608ae724b35b9785267ac61ddac7c146a59. You may find this interesting.